development/docker-compose.yml

```diff
@@ -30,6 +30,8 @@ services:
         - "http://localhost:8080/health/"
   celery_worker:
     image: "local/nautobot-dev:local-py${PYTHON_VER}"
+    ports:
+      - "8081:8080"
```

Comment:
How does this work in the case where there are multiple celery workers for a given Nautobot instance? Would you just collect metrics from one arbitrary worker, since I think each worker will have identical metrics based on this implementation?

Reply:
Excellent point. I'm going to write some thoughts about it here before I leave for the holidays.

# Multiple workers - one filesystem

In this case I think we should be good by wrapping a try/except around the `start_http_server` call as we don't really care which exact worker exposes the metrics and they all share the same filesystem directory (by the multiproc mechanism from the Python Prometheus client). This should work, because all of our workers should share the same lifetime - one wouldn't shut down significantly before another would, right?

# Multiple workers - multiple filesystems

In this case each worker will expose the metrics on its own port 8080, which the scraping configuration for a given Prometheus instance should be able to account for AFAIK.

Reply:
These are my findings here.

**One worker - one machine - one process**: No problem.
**One worker - one machine - multiple processes**: No problem as the server starts on worker startup rather than child process startup.
**Multiple workers - one machine - any amount of processes**: This would currently break with this implementation. Is this a scenario that has any practical relevance at all, do we need it?
**Multiple workers - one per machine - any amount of processes**: No problem - in this case every worker exposes their metrics on the configured port. Depending on which `JobResult` instance is processed where, the metrics would be generated on the corresponding worker. In the Prometheus scraping config we can then scrape all the workers as individual targets, Prometheus itself should take care of the appropriate aggregation.
**Multiple workers - multiple per machine - any amount of processes**: See "Multiple workers - one per machine - any amount of processes".

Reply:
I have addressed the cases where multiple workers run on one machine now, as well.